icubam/time_utils.py

```diff
@@ -54,29 +59,43 @@ def parse_hour(hour, sep=':') -> Tuple[str, str]:
     return "", ""
 
 
-def get_next_timestamp(hours, ts=None):
+def get_timestamp_for_timezone(d: datetime.datetime, tz=pytz.UTC) -> int:
+  """Localize the date with the timezone and returns its UTC timestamp."""
+  loc_d = tz.localize(d)
+  utc_d = loc_d - loc_d.utcoffset()
+  return utc_d.timestamp()
+
+
+def to_utc(d: datetime.datetime, tz=pytz.UTC) -> datetime.datetime:
+  """Translate a date from a given timezone to UTC."""
+  return tz.localize(d).astimezone(pytz.UTC)
+
+
+def get_next_timestamp(hours, ts=None, tz=pytz.UTC):
   """Gets the timestamp of the next hour in the list based on ts (or now).
 
   Args:
    hours: a list of tuples (hour, minute)
    ts: an optional timestamp. If none, use the current timestamp.
+   tz: a pytz.timezone to parse the the hours
   """
   if not hours:
     return None
 
-  ts = int(time.time()) if ts is None else ts
-  now = datetime.datetime.fromtimestamp(ts)
+  ts = utcnow() if ts is None else ts
+  now = to_utc(datetime.datetime.fromtimestamp(ts))
```

Comment:
This should already be in UTC if it comes from the DB.  Not sure what to_utc assumes.

Reply:
it s a bit more subtle indeed. what comes from the database is "naive", meaning, if i understand correctly, that it is utc based but it is not specified it is utc. In order to be able to compare two dates, it seems like if one is naive and one is localized, it fails. So here, the only thing this `to_utc` does is to say: specify the timezone to be utc, so that later, we will take the candidates datetimes (parsed from config) translate them into utc and compare to find the next one